### PR TITLE
Introduce MockNodeParameters/Args

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: node/src/test/kotlin/net/corda/node/services/network/AbstractNetworkMapServiceTest.kt [hunk 1]
```diff
@@ -267,13 +264,8 @@ abstract class AbstractNetworkMapServiceTest<out S : AbstractNetworkMapService>
...
     private object NoNMSNodeFactory : MockNetwork.Factory<MockNode> {
-        override fun create(config: NodeConfiguration,
-                            network: MockNetwork,
-                            networkMapAddr: SingleMessageRecipient?,
-                            id: Int,
-                            notaryIdentity: Pair<ServiceInfo, KeyPair>?,
-                            entropyRoot: BigInteger): MockNode {
-            return object : MockNode(config, network, null, id, notaryIdentity, entropyRoot) {
+        override fun create(args: MockNodeArgs): MockNode {
+            return object : MockNode(args.copy(networkMapAddr = null)) {
```
Comment (andr3ej) [inline node/src/test/kotlin/net/corda/node/services/network/AbstractNetworkMapServiceTest.kt]: we don't expect users to make hacks like this, so not adding support for them to MockNodeArgs is fine

File: testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt [hunk 2]
```diff
@@ -57,8 +56,7 @@ fun transaction(
 
 fun testNodeConfiguration(
         baseDirectory: Path,
-        myLegalName: CordaX500Name,
-        notaryConfig: NotaryConfig? = null): NodeConfiguration {
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt]: nothing used it

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt [hunk 3]
```diff
@@ -41,6 +36,13 @@ import java.util.concurrent.Future
 abstract class Simulation(val networkSendManuallyPumped: Boolean,
                           runAsync: Boolean,
                           latencyInjector: InMemoryMessagingNetwork.LatencyCalculator?) {
+    companion object {
+        private val defaultParams // The get() is necessary so that entropyRoot isn't shared.
+            get() = MockNodeParameters(configOverrides = {
+                doReturn(makeTestDataSourceProperties(it.myLegalName.organisation)).whenever(it).dataSourceProperties
```
Comment (andr3ej) [inline samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt]: enough to preserve existing behaviour, as this is the only thing MockNode changes from a stock testNodeConfiguration

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt [hunk 4]
```diff
@@ -117,33 +80,24 @@ abstract class Simulation(val networkSendManuallyPumped: Boolean,
...
     val mockNet = MockNetwork(
             networkSendManuallyPumped = networkSendManuallyPumped,
             threadPerNode = runAsync,
             cordappPackages = listOf("net.corda.irs.contract", "net.corda.finance.contract", "net.corda.irs"))
-    // This one must come first.
-    val notary = mockNet.createNotaryNode(validating = false, nodeFactory = NotaryNodeFactory)
-    val regulators = listOf(mockNet.createUnstartedNode(nodeFactory = RegulatorFactory))
-    val ratesOracle = mockNet.createUnstartedNode(nodeFactory = RatesOracleFactory)
-
+    // This one must come first. XXX: Really?
```
Comment (mnesbit) [inline samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt]: Yes I think we just do .first for oracles as we have no way of configuring them right now.

File: testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt [hunk 5]
```diff
@@ -68,7 +66,7 @@ fun testNodeConfiguration(
         doReturn("cordacadevpass").whenever(it).keyStorePassword
         doReturn("trustpass").whenever(it).trustStorePassword
         doReturn(emptyList<User>()).whenever(it).rpcUsers
-        doReturn(notaryConfig).whenever(it).notary
+        doReturn(null).whenever(it).notary
```
Comment (mnesbit) [inline testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt]: Why null?

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt (hunk 4 above)
Comment (andr3ej) [reply samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt]: this comment was on the now-removed network map, so i'm going to assume it's obsolete and remove it

File: testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt (hunk 5 above)
Comment (andr3ej) [reply testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt]: notaryConfig default value was null, so this preserves existing behaviour

**Comment by andr3ej** [comment]
@mnesbit i've checked Simulation.kt via Network Simulator and it works. there's an exception when running the simulation a second time but the same exception exists in master.

**Pull request merged**